### ISSUE TITLE
[0.4.3] react/no-unused-prop-types reports props forwarded through a JSX spread

### ISSUE THREAD

**Issue by fireairforce** [utooland/utoo-lint#1259]
## Summary

Props forwarded via <Child {...props} /> are used. utlint nevertheless reports value as unused when only other is directly destructured. Passing both props explicitly is accepted by both tools.

The original reproduction in #1253 is fixed in 0.4.3. This is an additional case, tracked separately. Verified on Node.js 22.23.1, macOS arm64.

## Setup

```sh
npm init -y
npm install --save-dev @utoo/lint@0.4.3 eslint@8.57.1 @typescript-eslint/parser@6.21.0 eslint-plugin-react@7.37.5
```

`.eslintrc.cjs`:

```js
module.exports = {
  "root": true,
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": 2022,
    "sourceType": "module",
    "ecmaFeatures": {
      "jsx": true
    }
  },
  "plugins": [
    "react"
  ],
  "rules": {
    "react/no-unused-prop-types": "error"
  },
  "settings": {
    "react": {
      "version": "18.2"
    }
  }
};
```

`utlint.config.json`:

```json
{
  "rules": {
    "react/no-unused-prop-types": "error"
  }
}
```

Run each case separately as `example.tsx`:

```sh
./node_modules/.bin/eslint --no-eslintrc --config .eslintrc.cjs --format=json example.tsx
./node_modules/.bin/utlint --config utlint.config.json --format=json example.tsx
```

## Reproduction

```tsx
interface Props{value:string;other:number;} export function Example(props:Props){const{other}=props;return <Child {...props} other={other}/>;}
```

ESLint:

```text
(no diagnostics)
```

utlint:

```text
'value' PropType is defined but prop is never used
```

## Control

```tsx
interface Props{value:string;other:number;} export function Example(props:Props){return <Child value={props.value} other={props.other}/>;}
```

ESLint:

```text
(no diagnostics)
```

utlint:

```text
(no diagnostics)
```

Only the displayed rule is enabled. Neither invocation has a parsing failure.
